src/main/java/org/mifos/connector/mpesa/camel/routes/PaybillRoute.java

```diff
@@ -1,106 +1,123 @@
-                    //Channel URL
-                    logger.debug("Channel URL : {}", channelUrl);
-                    e.setProperty("channelUrl", channelUrl);
+                    String businessShortCode = paybillRequestDTO.getShortCode();
+                    String amsName = mpesaPaybillProp.getAMSFromShortCode(businessShortCode);
+                    String currency = mpesaPaybillProp.getCurrencyFromShortCode(businessShortCode);
+                    String amsUrl = mpesaUtils.getAMSUrl(amsName);
+                    String accountHoldingInstitutionId = mpesaPaybillProp.getAccountHoldingInstitutionId();
+
+                    e.getIn().setHeader("amsUrl", amsUrl);
+                    e.getIn().setHeader("amsName", amsName);
+                    e.getIn().setHeader("accountHoldingInstitutionId", accountHoldingInstitutionId);
+                    e.setProperty("channelUrl", channelCamelUrl);
                     e.setProperty("secondaryIdentifier", secondaryIdentifierName);
                     e.setProperty("secondaryIdentifierValue", paybillRequestDTO.getMsisdn());
-                    e.setProperty("dfspId", dfspId);
                     ChannelRequestDTO obj = MpesaUtils.convertPaybillPayloadToChannelPayload(paybillRequestDTO, amsName, currency);
-                    return obj.toString();
+                    try {
+                        return objectMapper.writeValueAsString(obj);
+                    } catch (JsonProcessingException ex) {
+                        throw new RuntimeException(ex);
+                    }
                 })
                 .log("MPESA Request Body : ${body}")
                 .toD("${header.channelUrl}" + "/accounts/validate/${header.secondaryIdentifier}/${header.secondaryIdentifierValue}" + "?bridgeEndpoint=true&throwExceptionOnFailure=false")
-                .choice()
-                .when(header("CamelHttpResponseCode").isEqualTo("200"))
                 .log(LoggingLevel.INFO, "Request sent to channel")
+                .setBody(e -> {
+                    String paybillResponseBodyString = e.getIn().getBody(String.class);
+                    JSONObject paybillResponse = new JSONObject(paybillResponseBodyString);
+                    logger.debug("Paybill Response Body : {}", paybillResponseBodyString);
+                    logger.debug("Reconciled : {}", paybillResponse.getBoolean("reconciled"));
+                    GsmaTransfer gsmaTransfer = mpesaUtils.createGsmaTransferDTO(paybillResponse);
+                    e.getIn().removeHeaders("*");
+                    e.getIn().setHeader("accountHoldingInstitutionId", paybillResponse.getString("accountHoldingInstitutionId"));
+                    e.getIn().setHeader("amsName", paybillResponse.getString("amsName"));
+                    e.getIn().setHeader("Platform-TenantId", paybillResponse.getString("accountHoldingInstitutionId"));
+                    e.getIn().setHeader("X-CorrelationID", "12345678-6897-6798-6798-098765432134");
```

Comment:
create/reuse String constants for all the header keys.
donot hard code correlation ID